File: mapir-android-service-sdk/src/main/java/ir/map/servicesdk/MapirService.java
package ir.map.servicesdk;

import android.content.Context;
import android.os.Build;

import androidx.annotation.NonNull;

public final class MapirService {

    private static String _apiKey;
    private static String userAgent;

    public static void init(@NonNull Context context, @NonNull String apiKey) {
        if (apiKey != null && !apiKey.isEmpty()) {
            if (context != null) {
                _apiKey = apiKey;
                userAgent = userAgentString(context);
            } else throw new RuntimeException("Provide nonNull Context.");
        } else
            throw new RuntimeException("No APIKEY Provided, Please visit https://corp.map.ir/registration/ to get new APIKEY");
    }

    private static String userAgentString(@NonNull Context context) {
        return String.format("Android/%s(%s)(%s)-ServiceSdk/%s-%s(%s)/%s-(%s)",
                Build.VERSION.SDK_INT,
                Build.VERSION.RELEASE,
                Build.CPU_ABI,
                BuildConfig.MAPIR_SERVICE_SDK_VERSION,
                context.getPackageName(),
                getApplicationName(context),
                Build.BRAND,
                Build.MODEL
        );
    }

    static String getApiKey() {
        return _apiKey;
    }

    static String getUserAgent() {
        return userAgent;
    }

    private static String getApplicationName(Context context) {
        return context.getApplicationInfo().loadLabel(context.getPackageManager()).toString();
    }
}

File: mapir-android-service-sdk-ktx/src/main/java/ir/map/servicesdk/MapirService.kt
package ir.map.servicesdk

import android.content.Context
import android.os.Build

class MapirService {
    companion object{
        @JvmStatic
        var apiKey: String? = null
        var userAgent: String? = null

        @JvmStatic
        fun init(context: Context, apiKey: String) {
            if (apiKey != null && apiKey.isNotEmpty()) {
                if (context != null) {
                    Companion.apiKey = apiKey
                    userAgent = userAgentString(context)
                } else throw RuntimeException("Provide nonNull Context.")
            } else throw RuntimeException("No APIKEY Provided, Please visit https://corp.map.ir/registration/ to get new APIKEY")
        }

        @JvmStatic
        private fun userAgentString(context: Context) = String.format("Android/%s(%s)(%s)-ServiceSdk-ktx/%s-%s(%s)/%s-(%s)",
                Build.VERSION.SDK_INT,
                Build.VERSION.RELEASE,
                Build.CPU_ABI,
                BuildConfig.MAPIR_SERVICE_SDK_VERSION,
                context.packageName,
                getApplicationName(context),
                Build.BRAND,
                Build.MODEL
        )

        private fun getApplicationName(context: Context): String? {
            return context.applicationInfo.loadLabel(context.packageManager).toString()
        }
    }
}